### PR TITLE
Support for #if expression

### LINKED ISSUE #436: Support for #if expression
Support #if expression:

 * completion with data model
 * validation
 * higlight
 * definition
 * hover

### PR DESCRIPTION
**Pull request by angelozerr**
Support for #if expression

Fixes #436

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
Here a demo:

IfSupportDemo

File: qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteCompletionInExpressionTest.java [hunk 1]
```diff
@@ -1,3 +1,14 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
```
Comment (fbricon) [inline qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteCompletionInExpressionTest.java:2]: 2022 - Happy new year ;-)

Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteCompletionInExpressionTest.java:2]: This class has been written in 2021.

**Comment by fbricon** [comment]
thanks @angelozerr

**Pull request merged**